src/firefly/java/edu/caltech/ipac/firefly/server/servlets/AnyFileDownload.java

```diff
@@ -165,7 +165,16 @@ private static void logActivity(File f) {
     }
 
     private static void trackProgress(HttpServletRequest req, BackgroundEnv.DownloadProgress progress) {
-        StringKey statusKey= new StringKey(req.getQueryString());
+        String queryString =req.getQueryString();
+
+        if (queryString==null){
+            try {
+                throw new Exception("Status can not be tracked because the queryString is empty");
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
```

Comment:
I don't understand this block.  You throw an exception, then catch it to print its stacktrace.
Also, if you plan to log the error, use a logger, not standard out.